content/gotraining.slide

```diff
@@ -212,21 +224,279 @@ Execute with
 * Interlude: Towards Continuous Deployment
 
 
+* What is Continuous Deployment (CD)?
+
+Continuous Deployment != Continuous Delivery
+
+.image img/cd.jpg
+.caption [[https://blog.crisp.se/2013/02/05/yassalsundman/continuous-delivery-vs-continuous-deployment][Yassal Sundman, Continuous Delivery vs Continuous Deployment]]
+
+
+* Continuous Deployment Quotes
+
+"Continuous Deployment is a strategy for software releases wherein any code commit that passes the automated testing phase is automatically released into the production environment."
+[[https://searchitoperations.techtarget.com/definition/continuous-deployment][TechTarget definition]]
+
+
+"Continuous deployment requires a highly developed culture of monitoring, being on call, and having the capacity to recover quickly."
+[[https://semaphoreci.com/blog/2017/07/27/what-is-the-difference-between-continuous-integration-continuous-deployment-and-continuous-delivery.html][Marko Anastasov, Semaphore Engineering Blog]]
+
+
+* Recommendations for CD @ANZ (1/3)
+
+- Use *trunk* *based* *development* (not GitFlow)
+- Create *small* *PRs* (Pull Requests) that can be Squash Merged
+- ... which lead to revertible commits that have passed automated tests
+- Protect `master` branch on GitHub
+
+
+* Recommendations for CD @ANZ (2/3)
+
+Quality leads to the confidence to automatically deploy, so we need
+ways to increase and maintain quality.
+
+- Write great commit messages
+- Think about the reviewer and review your own PRs first
+- *Lint* strictly `golangci-lint` `run`
+- Ensure *100%* *coverage*
+
+
+* Recommendations for CD @ANZ (3/2)
+
+- Test (unit & system), lint & cover as part of *CI* (Continuous Integration)
+- *Version* all deployments (use [[https://semver.org/][semver]]; tag source, images etc.)
+- *Document* for newbies in the README.md (must!) and use godoc
+- Enjoy your CD as last step after successful CI run on master
+
+
+* Protect `master` branch on GitHub
```

Comment:
verbatim text (backquotes) doesn't seem to work in titles. Remove backquotes.

Reply:
I kinda did that on purpose, but if you think it's bad form I will remove it.

Reply:
Done.

Reply:
oh, too late now. if on purpose, you should have kept it. I do it on purpose too sometimes - it still gets the message across.